src/abstractdataframe/join.jl

```diff
@@ -416,9 +416,11 @@ function _join(df1::AbstractDataFrame, df2::AbstractDataFrame;
     end
 
     if indicator !== nothing
-        refs = left_indicator + right_indicator
-        pool = CategoricalPool{String,UInt8}(["left_only", "right_only", "both"])
-        indicatorcol = CategoricalArray{String,1}(refs, pool)
+        left_indicator .+= right_indicator
+        pa_base = PooledArray(["left_only", "right_only", "both"])
+        indicatorcol = PooledArray(PooledArrays.RefArray(left_indicator),
+                                   Dict{String, UInt8}("left_only" => 0x1, "right_only" => 0x2, "both" => 0x3),
```

Comment:
BTW, I've just checked an CSV uses `UInt32`:
https://github.com/JuliaData/CSV.jl/blob/2e8a8f38aba186161ff4eeaeada9756457f010ea/src/file.jl#L424

It could make sense to use that even if that wastes space since it will reduce latency. Though not a big deal.

Reply:
I thought about it, and decided that having smaller footprint is more important. But OK, I will change it.